packages/tss/src/tss.ts

```diff
@@ -0,0 +1,590 @@
+import { decrypt, ecCurve, encrypt, getPubKeyPoint, KeyDetails, LocalMetadataTransitions, Point, ShareStore, TKeyArgs } from "@tkey/common-types";
+import ThresholdKey, { CoreError, lagrangeInterpolation, Metadata } from "@tkey/core";
+import { generatePrivate } from "@toruslabs/eccrypto";
+import { dotProduct, ecPoint, getLagrangeCoeffs, hexPoint, PointHex, randomSelection, RSSClient } from "@toruslabs/rss-client";
+import BN from "bn.js";
+import { keccak256 } from "ethereum-cryptography/keccak";
+
+import { TSSTorusServiceProvider } from ".";
+import { FactorEnc, IAccountSaltStore, InitializeNewTSSKeyResult } from "./common";
+import { generateSalt, kCombinations } from "./util";
+
+export const TSS_MODULE = "tssModule";
+
+export interface TSSTKeyArgs extends TKeyArgs {
+  serviceProvider: TSSTorusServiceProvider;
```

Comment:
so we ovewrite the serviceProvider type rite.

Reply:
yes, this is intentional: `TSSTorusServiceProvider` extends `TorusServiceProvider`, so the core can use this object as before. At the same time, `TKeyTSS` can use the added functionality like `getRSSNodeDetails` and so on.